lib/postgrex/protocol.ex

```diff
@@ -537,16 +540,19 @@ defmodule Postgrex.Protocol do
           {:ok, Postgrex.Result.t(), state}
           | {DBConnection.status(), state}
           | {:disconnect, %RuntimeError{}, state}
-          | {:disconnect, %DBConnection.ConnectionError{} | Postgrex.Error.t(), state}
+          | {:disconnect,
+             %DBConnection.ConnectionError{} | Postgrex.Error.t() | Postgrex.QueryError.t(),
```

Comment:
Let's return Postgrex.Error. It is not worth introducing a new exception type only for this case. WDYT? :)

Reply:
Fine with me!

Reply:
We can also return a RuntimeError if you prefer.

Reply:
Went with Postgrex.Error in https://github.com/elixir-ecto/postgrex/pull/669/commits/af319f9ad753023e180ec7883cd16b39674711aa